src/commands/dsyms/utils.ts

```diff
@@ -0,0 +1,127 @@
+import {exec} from 'child_process'
+import {mkdirSync} from 'fs'
+import glob from 'glob'
+import path from 'path'
+import {promisify} from 'util'
+import {buildPath} from '../../helpers/utils'
+import {Payload} from './interfaces'
+import {checkNonEmptyFile} from './validation'
+
+const UUID_REGEX = '[0-9A-F]{8}-([0-9A-F]{4}-){3}[0-9A-F]{12}'
+
+const getMatchingDSYMFiles = async (absoluteFolderPath: string): Promise<Payload[]> => {
+  const dSYMFiles = glob.sync(buildPath(absoluteFolderPath, '**/*.dSYM'))
+
+  return Promise.all(
+    dSYMFiles.map(async (dSYMPath) => {
+      const uuids = await dwarfdumpUUID(dSYMPath)
+
+      return {
+        path: dSYMPath,
+        type: 'ios_symbols',
+        uuids,
+      }
+    })
+  )
+}
+
+export const getPayloads = async (searchPaths: string[]): Promise<Payload[]> => {
+  const nestedPayloads = await Promise.all(searchPaths.map((searchPath: string) => getMatchingDSYMFiles(searchPath)))
+
+  return nestedPayloads.flat()
+}
+
+export const getSearchPaths = async (baseSearchPath: string): Promise<string[]> => {
+  const zipFiles = getZipFiles(baseSearchPath)
+  const unzipFolder = buildPath('/tmp', 'datadog-ci', 'dsyms')
+  const searchPaths = (await Promise.all(
+    zipFiles
+      .map(async (zipFile: string) => {
+        const unzipPath = buildPath(unzipFolder, path.basename(zipFile, '.zip'), Date.now().toString())
+        try {
+          await unzip(zipFile, unzipPath)
+
+          return unzipPath
+        } catch (err) {
+          console.log(err)
+        }
+      })
+      .filter((item) => item !== undefined)
+  )) as string[]
```

Comment:
Why do you need to do this?

Reply:
i couldn't get TS compiler to compile without this, otherwise TS sees the resulting array as `[string | undefined]` type whereas it should be `[string]` despite `filter((item) => item !== undefined)`

i'll be more than happy to solve this in a better way, open to suggestions

Reply:
I think it might come from the `console.log(err)`, try to `return []` right after it.